homeassistant/components/apsystems/config_flow.py

```diff
@@ -17,35 +19,35 @@
 )
 
 
-class APsystemsLocalAPIFlow(config_entries.ConfigFlow, domain=DOMAIN):
+class APsystemsLocalAPIFlow(ConfigFlow, domain=DOMAIN):
     """Config flow for Apsystems local."""
 
     VERSION = 1
 
     async def async_step_user(
-        self,
-        user_input: dict | None = None,
-    ) -> config_entries.ConfigFlowResult:
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
         """Handle a flow initialized by the user."""
-        _errors = {}
-        session = async_get_clientsession(self.hass, False)
+        errors = {}
 
         if user_input is not None:
+            session = async_get_clientsession(self.hass, False)
+            api = APsystemsEZ1M(user_input[CONF_IP_ADDRESS], session=session)
             try:
-                session = async_get_clientsession(self.hass, False)
-                api = APsystemsEZ1M(user_input[CONF_IP_ADDRESS], session=session)
                 device_info = await api.get_device_info()
-                await self.async_set_unique_id(device_info.deviceId)
-            except (TimeoutError, client_exceptions.ClientConnectionError) as exception:
-                LOGGER.warning(exception)
-                _errors["base"] = "connection_refused"
+            except (TimeoutError, client_exceptions.ClientConnectionError):
+                LOGGER.exception("Could not connect")
```

Comment:
Even though this is cleaner, I am also wondering what the added value is of having a logger here as the exceptions catched are connection exceptions, so what extra info does this give you?